scripts/install

```diff
@@ -6,219 +6,224 @@
-ynh_replace_string --match_string "\"FeedbackName\": \"\""            --replace_string "\"FeedbackName\": \"Mattermost notification\""     --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"FeedbackEmail\": \"\""           --replace_string "\"FeedbackEmail\": \"no-reply@${domain}\""         --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"SMTPUsername\": \"\""            --replace_string "\"SMTPUsername\": \"${mattermost_user}\""          --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"SMTPPassword\": \"\""            --replace_string "\"SMTPPassword\": \"${mattermost_user_password}\"" --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"SMTPServer\": \"\""              --replace_string "\"SMTPServer\": \"localhost\""                     --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"SMTPPort\": \".*\""              --replace_string "\"SMTPPort\": \"25\""                              --target_file $final_path/config/config.json
-# Disable Mattermost debug console by default
-ynh_replace_string --match_string "\"EnableConsole\": true"           --replace_string "\"EnableConsole\": false"                          --target_file $final_path/config/config.json
-# Configure log file location
-ynh_replace_string --match_string "\"FileLocation\": \"\""            --replace_string "\"FileLocation\": \"$logs_path\""                  --target_file $final_path/config/config.json
-# Disable questionable password complexity rules
-ynh_replace_string --match_string "\"MinimumLength\": 10"             --replace_string "\"MinimumLength\": 8"                              --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"Lowercase\": true"               --replace_string "\"Lowercase\": false"                              --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"Number\": true"                  --replace_string "\"Number\": false"                                 --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"Uppercase\": true"               --replace_string "\"Uppercase\": false"                              --target_file $final_path/config/config.json
-ynh_replace_string --match_string "\"Symbol\": true"                  --replace_string "\"Symbol\": false"                                 --target_file $final_path/config/config.json
-# Configure analytics according to user choice
-if [ $analytics -eq 0 ]; then
-    ynh_replace_string --match_string "\"EnableDiagnostics\": true"   --replace_string "\"EnableDiagnostics\": false"                      --target_file $final_path/config/config.json
-fi
-ynh_app_setting_set "$app" analytics "$analytics"
+cp ../conf/config.json $final_path/config/config.json
```

Comment:
If we have to patch the config file anyway, I think it may be better to copy it from the downloaded source (rather than including an outdated copy in `../conf`).

Reply:
Patching the downloaded config file seems like a good idea in terms of maintenance time, but it's not because the settings can change without warning and break everything silently. I think that's what caused the script to fail in the first place. From version 5.30.1 (not documented in the changelog), the default database is set to PostgreSQL (See L149 `"DriverName": "postgres",`).  this value was not replaced in the install script ...

5.29.1
```
"DriverName": "mysql",
"DataSource": "mmuser:mostest@tcp(localhost:3306)/mattermost_test?charset=utf8mb4,utf8\u0026readTimeout=30s\u0026writeTimeout=30s",
```

5.30.1
```
"DriverName": "postgres",
"DataSource": "postgres://mmuser:mostest@localhost/mattermost_test?sslmode=disable\u0026connect_timeout=10",
```

to add to the madness, there are 2 `DriverName` entries in the configuration file L.146 for the `SqlSettings` and L.203 for the `FileSettings` ...

Reply:
Hmm, that's right.

Well, in the end we could patch the config file using `jq` to properly find the JSON path (instead of relying on the key name).

But I agree, the benefits of not including the config file are not that clear. I'm fine with your approach (and we can switch to something else later if it proves complicated to maintain).